src/vim9type.c

```diff
@@ -1886,38 +1888,21 @@ check_typval_is_value(typval_T *tv)
     int
 check_type_is_value(type_T *type)
 {
+    if (type == NULL)
+	return OK;
     if (type->tt_type == VAR_CLASS)
     {
         semsg(_(e_using_class_as_value_str), type->tt_class->class_name);
 	return FAIL;
     }
     else if (type->tt_type == VAR_TYPEALIAS)
     {
-	// Not sure what could be done here to get a name
-	// TODO: MAYBE AN OPTIONAL ARGUMENT
+	// TODO: Not sure what could be done here to get a name.
+	//       Maybe an optional argument?
```

Comment:
The typealias_S structure has the typealias name.  A pointer to this is currently not stored in type_S. For a class, it is stored tt_class.  We need to add a variable (tt_typealias) to type_S to get to the name.

Reply:
Since you're requesting a change, I guess you want this change made.

I can do it, but I need some guidance. Where does this change go?

Reply:
And should `tt_class`/`tt_typealias` be in a union?

Reply:
There's also
```
vim9script
class C
endclass
type T = C
def F()
    abs(T)
enddef
F()
```
which outputs
```
E1405: Class "C" cannot be used as a value
```
Would/could adding a field "tt_typealias" help give an accurate message when using a typealias for a class?

Reply:
> Since you're requesting a change, I guess you want this change made.
> 
> I can do it, but I need some guidance. Where does this change go?

This change should be done in a separate follow-on PR.

Reply:
> > Since you're requesting a change, I guess you want this change made.
> > I can do it, but I need some guidance. Where does this change go?
> 
> This change should be done in a separate follow-on PR.

OK. I'll do "re-request review" to clear the "change requested" flag. And we can go from there.

Reply:
> The typealias_S structure has the typealias name. A pointer to this is currently not stored in type_S. For a class, it is stored tt_class. We need to add a variable (tt_typealias) to type_S to get to the name.

After looking into this some more, I realized that `type->tt_type` will never be set to VAR_TYPEALIAS.  In parse_type(), when parsing a type alias, the associated aliased type is returned (line 1402 in vim9type.c).

Reply:
I opened #13696 yesterday about having better error messages.